File: Homeworks/C#2/StringsAndTextProcessingHomework/05.ParseTags/ParseTags.cs
/*
Problem 5. Parse tags

You are given a text. Write a program that changes the text in all regions surrounded by the tags <upcase> and </upcase> to upper-case.
The tags cannot be nested.
Example: We are living in a <upcase>yellow submarine</upcase>. We don't have <upcase>anything</upcase> else.

The expected result: We are living in a YELLOW SUBMARINE. We don't have ANYTHING else.
*/ 


using System;

namespace ParseTagsProblem
{
    class ParseTags
    {

        static string ParseTagsMethod(string text)
        {
            string result = string.Empty;

            for (int i = 0; i < text.Length; i++)
            {
                if (text[i] == '<')
                {
                    i += 8;
                    while (text[i] != '<')
                    {
                        result += text[i].ToString().ToUpper();
                        i++;
                    }

                    i += 8;
                }
                else
                {
                    result += text[i];
                }
            }

            return result;
        }

        static void Main()
        {
            Console.WriteLine("Please, enter some text: ");
            string inputText = Console.ReadLine();

            Console.WriteLine(ParseTagsMethod(inputText));
        }
    }
}

File: Homeworks/JavaScriptFundamentals/08.Strings/04.ParseTags/ParseTags.js
/*
 You are given a text. Write a function that changes the text in all regions:

 <upcase>text</upcase> to uppercase.
 <lowcase>text</lowcase> to lowercase
 <mixcase>text</mixcase> to mix casing(random)
 Example: We are <mixcase>living</mixcase> in a <upcase>yellow submarine</upcase>.
 We <mixcase>don't</mixcase> have <lowcase>anything</lowcase> else.

 The expected result:
 We are LiVinG in a YELLOW SUBMARINE. We dOn'T have anything else.

 Note: Regions can be nested.
*/

var jsConsole,
    text = 'We are <mixcase>living</mixcase> in a <upcase>yellow submarine</upcase>.' +
    'We <mixcase>don\'t</mixcase> have <lowcase>anything</lowcase> else.',
    result = '';

for (var i = 0; i < text.length; i += 1) {

    if (text[i] == '<' && text[i + 1] == 'u') {
        i += 8;
        while (text[i] != '<' ) {
            result += text[i].toUpperCase();
            i += 1;
        }
        if(text[i + 1] == '/') {
            i += 8;
        }
        else {
            i--;
        }
    }
    else if(text[i] == '<' && text[i + 1] == 'm'){
        i += 9;
        while (text[i] != '<'){
            if(Math.random() < 0.5){
                result += text[i].toUpperCase();
                i += 1;
            }
            else{
                result += text[i].toLowerCase();
                i += 1;
            }
        }
        if(text[i + 1] == '/') {
            i += 9;
        }else {
            i--;
        }
    }
    else if (text[i] == '<' && text[i + 1] == 'l') {
        i += 9;
        while (text[i] != '<') {
            result += text[i].toLowerCase();
            i += 1;
        }
        if(text[i + 1] == '/'){
            i += 9;
        } else {
            i--;
        }
    }

    else {
        result += text[i];
    }
}


jsConsole.writeLine(text);
jsConsole.writeLine(result);

console.log(text);
console.log(result);